packages/server/src/assertions/assert.ts

```diff
@@ -0,0 +1,18 @@
+export class ServiceAssertionError extends Error {
+  constructor(message: string) {
+    super(`INTERNAL ASSERTION FAILURE:
+    While running BigTest, ${message} was received.
+    this is almost certainly a bug in BigTest, and we would be very grateful if you would report
+    it at https://github.com/thefrontside/bigtest/issues/new
+    `)
+
+    Object.setPrototypeOf(this, new.target.prototype);
+  }
+}
+
+export function assert(condition: unknown, msg: string): asserts condition {
+  if (!condition) {
+    throw new ServiceAssertionError(msg);
+  }
+}
+
```

Comment:
I don't see this actually used anywhere?

Reply:
Sorry, I've added it to `ManifestGeneratorOptions`.  It type narrows

```ts
  if(!options.files) {
    throw new Error('no files options in ManifestGeneratorOptions');
  }

  if(!options.destinationPath) {
    throw new Error('no files options in ManifestGeneratorOptions');
  }
```

to

```ts
  assert(options.files, 'no files options in ManifestGeneratorOptions');
  assert(options.destinationPath, 'no destinationApth in ManifestGeneratorOptions');
```